website/server/controllers/api-v3/chat.js

```diff
@@ -141,6 +158,49 @@ api.postChat = {
 
     let group = await Group.getGroup({user, groupId});
 
+    // Check message for banned slurs
+    if (textContainsBannedSlur(req.body.message)) {
+      let message = req.body.message;
+      user.flags.chatRevoked = true;
+      await user.save();
+
+      // Email the mods
+      let authorEmail = user.auth.local.email;
```

Comment:
Try using the `getUserInfo` function to get the email address, from `website/server/libs/email.js`. It will fetch addresses for Facebook and Google sign-in users when needed.

This might stop the InternalServerError you were getting (e.g., when there's no `local` authentication email address).